language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/AbstractDocumentServiceContext.java

```diff
@@ -132,47 +128,30 @@ public LSOperation operation() {
 
     @Override
     public List<Symbol> visibleSymbols(Position position) {
```

Comment:
Is avoiding repetition the reason to keep this without deprecating?  If so shall we make this method protected?

Reply:
Didn't get you?

Reply:
We discussed that we don't want this method to act as an interface for the semantic model. So I was wondering why we are not removing this method? (Is it to avoid repetition of logic to access and retrieve the visible symbols in child classes? ).
Each time this method is invoked, we would be calling the visibleSymbols API of the semantic model. So, we should either cache the visible symbols in the class or make this method protected so that this is not publically available. WDYT?

Reply:
Yes, that's correct. We can/should remove this method. But, since this is an interface, we cannot remove a method. What we can do is first deprecate this method and then remove it in a future release. If we are to deprecate, first we have to deprecate the `visibleSymbols()` method in the topmost interface and then add that to the `PositionedOperationContext`. That's why I decided not to remove it and keep it instead.

Reply:
+1. Let's add the deprecated annotation

Reply:
+1. Let's keep it